### DISCUSSION TITLE
failed to add cluster, error: x509.UnknownAuthorityError

### DISCUSSION THREAD

**Discussion by songminglong** [kubevela/kubevela#3013] [Q&A] [answered]
vela version: v1.2.0-beta.2

steps:
- vela cluster join admin.kubeconfig
`Error: failed to ensure resourcetracker crd installed in cluster ntgxh-x: failed to check resourcetracker crd in cluster ntgxh-x: x509: certificate signed by unknown authority`

cluster-gateway：
`E1228 02:25:35.818771       1 status.go:71] apiserver received an error that is not an metav1.Status: x509.UnknownAuthorityError{Cert:(*x509.Certificate)(0xc0009dd080), hintErr:error(nil), hintCert:(*x509.Certificate)(nil)}: x509: certificate signed by unknown authority`

我使用的是minikube，证书都是自签名的，包括内部集群也是自签名证书，但是增加cluster的时候，报x509错误。请问如何避免这种错误？如果必须使用有效的第三方ca，那么很多自签名的环境就没法玩了

**Comment by songminglong** [comment, marked as answer]
apiVersion: v1
clusters:
- cluster:
    certificate-authority: /root/kubevela/install/ca.crt
    server: https://xxx:443
  name: ntgxh-x
contexts:
- context:
    cluster: ntgxh-x
    user: default-user
  name: default-context
current-context: default-context
kind: Config
preferences: {}
users:
- name: default-user
  user:
    client-certificate: /root/kubevela/install/admin.crt
    client-key: /root/kubevela/install/admin.key

**Reply by Somefive** [reply]
Currently, we only support tls.crt, tls.key, ca.crt in user fields for X509 mode. The client-certificate and client-key are not supported now. You can try to embed the content of these certificate files in kubeconfig directly.

**Reply by songminglong** [reply]
已解决
